docker-compose/1_Auto_Upstall/noco.sh

```diff
@@ -45,486 +46,538 @@ print_warning() { print_color "$YELLOW" "WARNING: $1"; }
+			echo "$ip"
+			return
+		fi
+	fi
+
+	# Method 2: Using curl
+	if command -v curl >/dev/null 2>&1; then
+		ip=$(curl -s -4 https://ifconfig.co 2>/dev/null)
+		if [ -n "$ip" ]; then
+			echo "$ip"
+			return
+		fi
+	fi
+
+	# Method 3: Using wget
+	if command -v wget >/dev/null 2>&1; then
+		ip=$(wget -qO- https://ifconfig.me 2>/dev/null)
+		if [ -n "$ip" ]; then
+			echo "$ip"
+			return
+		fi
+	fi
+
+	# Method 4: Using host
+	if command -v host >/dev/null 2>&1; then
+		ip=$(host myip.opendns.com resolver1.opendns.com 2>/dev/null | grep "myip.opendns.com has" | awk '{print $4}')
+		if [ -n "$ip" ]; then
+			echo "$ip"
+			return
+		fi
+	fi
+
+	# If all methods fail, return localhost
+	echo "localhost"
 }
 
 get_nproc() {
-    # Try to get the number of processors using nproc
-    if command -v nproc &> /dev/null; then
-        nproc
-    else
-        # Fallback: Check if /proc/cpuinfo exists and count the number of processors
-        if [[ -f /proc/cpuinfo ]]; then
-            grep -c ^processor /proc/cpuinfo
-        # Fallback for macOS or BSD systems using sysctl
-        elif command -v sysctl &> /dev/null; then
-            sysctl -n hw.ncpu
-        # Default to 1 processor if everything else fails
-        else
-            echo 1
-        fi
-    fi
+	# Try to get the number of processors using nproc
+	if command -v nproc &>/dev/null; then
+		nproc
+	else
+		# Fallback: Check if /proc/cpuinfo exists and count the number of processors
+		if [[ -f /proc/cpuinfo ]]; then
+			grep -c ^processor /proc/cpuinfo
+		# Fallback for macOS or BSD systems using sysctl
+		elif command -v sysctl &>/dev/null; then
+			sysctl -n hw.ncpu
+		# Default to 1 processor if everything else fails
+		else
+			echo 1
+		fi
+	fi
 }
 
-
 prompt() {
-    local prompt_text="$1"
-    local default_value="$2"
-    local response
-
-    if [ -n "$default_value" ]; then
-        prompt_text+=" (default: $default_value)"
-    fi
-    prompt_text+=": "
-
-    read -r -p "$prompt_text" response
-    if [ -z "$response" ] && [ -n "$default_value" ]; then
-        echo "$default_value"
-    else
-        echo "$response"
-    fi
+	local prompt_text="$1"
+	local default_value="$2"
+	local response
+
+	if [ -n "$default_value" ]; then
+		prompt_text+=" (default: $default_value)"
+	fi
+	prompt_text+=": "
+
+	read -r -p "$prompt_text" response
+	if [ -z "$response" ] && [ -n "$default_value" ]; then
+		echo "$default_value"
+	else
+		echo "$response"
+	fi
+}
+
+prompt_oneof() {
+	local response
+	local resp_upper
+	local oneof_text
+	local prompt_text="$1"
+	local default_response="$2"
+	shift 1
+
+	prompt_text+=" (default: $default_response) "
+	oneof_text+="["
+	for one in "$@"; do
+		oneof_text+="$one,"
+	done
+	oneof_text="${oneof_text%%,}]"
+	prompt_text+="${oneof_text}: "
+
+	while true; do
+		read -r -p "$prompt_text" response
+		if [ -z "$response" ]; then
+			echo "$default_response"
+			return
+		fi
+
+		for one in "$@"; do
+			resp_upper="${response^^}"
```

Comment:
Looks like this syntax is supported widely

<img width="435" alt="Screenshot 2024-09-30 at 12 27 03 PM" src="https://github.com/user-attachments/assets/f85cb9bb-e305-4575-ae8c-979df24e5a08">


so let's replace it with something like following which

```bash
response_upper=$(echo "$response" | tr '[:lower:]' '[:upper:]')
```